tests/test-write-diagnostics.c

```diff
@@ -0,0 +1,287 @@
+/* -*- mode: C; c-file-style: "gnu"; indent-tabs-mode: nil; -*- */
+/*
+ * Copyright © 2018 Endless Mobile, Inc.
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, see <http://www.gnu.org/licenses/>.
+ */
+#include "config.h"
+#include <string.h>
+#include <locale.h>
+
+#include <glib.h>
+#include <glib/gstdio.h>
+
+#include "gis-write-diagnostics.h"
+#include "glnx-fdio.h"
+#include "glnx-shutil.h"
+
+typedef struct {
+  gchar *tmpdir;
+  gchar *image_dir;
+  gchar *home_dir;
+
+  gchar *output;
+  GError *error;
+} Fixture;
+
+static void
+fixture_set_up (Fixture *fixture,
+                gconstpointer user_data)
+{
+  GError *error = NULL;
```

Comment:
Nitpick: `g_autoptr()` for consistency and documentation of ownership, even if we will always abort if the error is ever set.